bashpp

```diff
@@ -1,18 +1,20 @@
 #!/usr/bin/env bash
 set -e
 
+#include _package/oci
+#include _package/ghrelease
```

Comment:
Here's the rationale:

* Modularity: Splitting these functions prepares Bash+ for future scalability. As more backends (e.g., local archives, SCMs, package managers) are added, this modular design allows for selective inclusion and avoids monolithic scripts.
* Bootstrapping: Like compilers such as GCC, Bash+ precompiles itself into monolithic scripts. It's consistent with the project's goals to have source and compiled files coexist.
* Future Proofing: This design aligns with Bash+'s modularity vision. Users can compile preprocessors with only the desired backends, reducing unnecessary overhead.

I understand the concern about duplication and "circular dependencies". However, having source files alongside the monolithic compiled output is an intentional feature of Bash+, designed for maintainability and extensibility.

The Makefile, the pre-commit configuration, and the GitHub actions workflow mostly automate the compilation process, ensuring that only compiled files are getting published or installed by Basher.

Reply:
- Modularity: I understand the organizational rationale for splitting the backends into multiple files, and I'm inclined to agree. However, this could be more easily accomplished with a simple one-line for loop: `for file in backends/*; do source "$file";done`. This would bypass the duplication and circular dependency with no discernible downside.
- Bootstrapping: Can you explain the reasoning for source and compiled files coexisting in source control being a project goal?
- Future Proofing: This is essentially the same point as Modularity; see above.

Can you please expand on how having source files alongside the monolithic compiled output is a positive for maintainability, and not a negative?